ipa-core/src/cli/hpke.rs

```diff
@@ -0,0 +1,337 @@
+use std::{
+    fmt::Debug,
+    fs::{read_to_string, File, OpenOptions},
+    io::{BufRead, BufReader, Write},
+    iter::zip,
+    path::PathBuf,
+};
+
+use clap::Parser;
+use rand::rngs::StdRng;
+use rand_core::SeedableRng;
+
+use crate::{
+    cli::playbook::InputSource,
+    config::{hpke_registry, HpkeServerConfig, KeyRegistries, NetworkConfig},
+    error::BoxError,
+    ff::{
+        boolean_array::{BA20, BA3, BA8},
+        U128Conversions,
+    },
+    hpke::{KeyRegistry, PrivateKeyOnly},
+    report::{EncryptedOprfReport, EventType, OprfReport, DEFAULT_KEY_ID},
+    secret_sharing::IntoShares,
+    test_fixture::{ipa::TestRawDataRecord, Reconstruct},
+};
+
+type BreakdownKey = BA8;
```

Comment:
I had a feeling that these values are hardcoded somewhere else. It would be better if we use the same hardcoded types across the whole IPA stack, so we can fix/change them in one place

Reply:
yea, that makes sense. they weren't public where I found them, but I can make them public.